index.js

```diff
@@ -3,8 +3,13 @@
  * @typedef {import('hast').Properties} Properties
  * @typedef {import('hast').Element['children'][number]} ElementChild
  *
+ * @typedef {Options | ((node: import('hast').Element, tree: Root) => Options)} inputOptions
+ *   Configuration as either a function or an object.
+ *
  * @typedef Options
  *   Configuration.
+ * @property {string[]} [exclude=[]]
+ *   Exclude anchors with the following attributes from `rehype-external-links`
```

Comment:
```suggestion
```

I was under the impression that *just* a function form of `options.target`, `options.rel`, `options.content` would be enough, and that the same functionality of `exclude` can be achieved with them?

Reply:
Here’s what kind of types I was thinking off:

```js
/**
 * @typedef {'_self'|'_blank'|'_parent'|'_top'} Target
 * @typedef {Array<string>|string} Rel
 * @typedef {ElementChild|Array<ElementChild>} Content
 * @typedef {Properties} ContentProperties
 *
 * @callback TargetCallback
 * @param {Element} node
 * @returns {Target|undefined}
 *
 * @callback RelCallback
 * @param {Element} node
 * @returns {Rel|undefined}
 *
 * @callback ContentCallback
 * @param {Element} node
 * @returns {Content|undefined}
 *
 * @callback ContentPropertiesCallback
 * @param {Element} node
 * @returns {Properties|undefined}
 *
 * ...
 * 
 * @property {Target|TargetCallback|false} [target='_blank']
 *   ...
 * @property {Rel|RelCallback|false} [rel=['nofollow', 'noopener', 'noreferrer']]
 *   ...
 * @property {Content|ContentCallback} [content]
 *   ...
 * @property {ContentProperties|ContentPropertiesCallback} [contentProperties]
 *   ...
 */
 ```

Reply:
I'll take a look

Reply:
I'll move the `exclude` config option to a new pr